src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/PartiallyLoweredLocalFunctionReference.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This represents a partially lowered local function reference (e.g.,
+    /// a local function call or delegate conversion) with relevant proxies
+    /// attached. It will later be rewritten by the
+    /// <see cref="LambdaRewriter.LocalFunctionReferenceRewriter"/> into a
+    /// proper call.
+    /// </summary>
+    internal class PartiallyLoweredLocalFunctionReference : BoundExpression
```

Comment:
@cston This node is as hidden as possible to try to prevent errors like that 😄 
It's a very short-lived node, only existing between the two passes of Lambda lowering.

I'm not opposed to throwing `Unreachable` for Kind, though.